src/runtime/auto-prompt-manager.ts

```diff
@@ -214,6 +223,50 @@ export class AutoPromptManager {
       return;
     }
     // Set experiment flags here.
+    const articleExpFlags =
+      this.entitlementsManager_.parseArticleExperimentConfigFlags(article);
+    this.onsitePreviewEnabled_ = articleExpFlags.includes(
+      ArticleExperimentFlags.ONSITE_PREVIEW_ENABLED
+    );
+  }
+
+  /**
+   * Displays the appropriate auto prompt for onsite preview.
+   */
+  private async showPreviewAutoPrompt_(
+    article: Article,
+    params: ShowAutoPromptParams
+  ): Promise<void> {
+    const actions = article.audienceActions?.actions;
+    if (!actions || actions.length === 0) {
+      return;
+    }
+    // Article response is honored over code snippet in case of conflict, such
+    // as when publisher changes revenue model but does not update snippet.
+    this.autoPromptType_ = this.getAutoPromptType_(
+      article.audienceActions?.actions,
+      params.autoPromptType
+    )!;
+
+    // Default isClosable to what is set in the page config.
+    // Otherwise, the prompt is blocking for publications with a
+    // subscription revenue model, while all others can be dismissed.
+    this.isClosable_ = params.isClosable ?? !this.isSubscription_();
+
+    const potentialAction = actions[0];
+
+    const promptFn = this.isMonetizationAction_(potentialAction?.type)
```

Comment:
Can we make it as a function if it's a shared logic between both flows?

Reply:
Done